src/allencell_ml_segmenter/main/viewer.py

```diff
@@ -105,3 +116,47 @@ def _get_layer_by_name(self, name: str) -> Optional[Layer]:
             if l.name == name:
                 return l
         return None
+
+    def get_seg_layers(self, layer_list: list[Layer]) -> list[Layer]:
+        """
+        Get only segmentation layers (which should be probability mappings) from the viewer.
+        These are the layers that start with [seg].
+        """
+        return [
+            layer
+            for layer in self.get_layers()
+            if layer.name.startswith("[seg]")
+        ]
+
+    def insert_threshold(
+        self, layer_name: str, image: np.ndarray, seg_layers: bool = False
+    ) -> None:
+        """
+        Insert a thresholded image into the viewer.
+        If a layer for this thresholded image already exists, the new image will replace the old one and refresh the viewer.
+        If the layer does not exist, it will be added to the viewer in the correct place (on top of the original segmentation image:
+        index_of_segmentation + 1 in the LayerList)
+        """
+        layer_to_insert = self._get_layer_by_name(f"[threshold] {layer_name}")
+        if layer_to_insert is None:
+            # No thresholding exists, so we add it to the correct place in the viewer
+            layerlist = self.viewer.layers
+
+            # check if the original segementation layer is currently in the viewer, if so, remove later after
+            # thresholding is applied
+            seg_layer_og: Optional[Layer] = None
+            if seg_layers:
+                seg_layer_og = self._get_layer_by_name(layer_name)
+
+            # figure out where to insert the new thresholded layer (on top of the original segmentation image)
+            layerlist_pos = layerlist.index(layer_name)
+            labels_created = Labels(image, name=f"[threshold] {layer_name}")
+            layerlist.insert(layerlist_pos + 1, labels_created)
+
+            # remove the original segmentation layer if it exists
+            if seg_layer_og:
```

Comment:
so, `seg_layers` is a flag for whether to remove corresponding segmentation layers after thresholding? Maybe rename to `remove_seg_layers`?